### PR TITLE
ToTableMapping (mapping of nested lists)

### PR DESCRIPTION
**Pull request by jakobbraun**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathIteratorFactory.java [hunk 1]
```diff
@@ -32,4 +34,13 @@ public DocumentPathIteratorFactory(final DocumentPathExpression path, final Docu
...
         }
     }
+
+    /**
+     * Gives a stream of iteration states.
+     * 
+     * @return stream of {@link PathIterationStateProvider}
+     */
+    public Stream<PathIterationStateProvider> stream() {
+        return StreamSupport.stream(spliterator(), false);
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathIteratorFactory.java:44]: Do you have to override the `splititerator` or the default from iterator okay here?

File: src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMapper.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * This call maps document data to Exasol rows according to {@link SchemaMappingQuery}.
+ */
+@java.lang.SuppressWarnings("squid:S119") // DocumentVisitorType does not fit naming conventions.
+public class SchemaMapper<DocumentVisitorType> {
+    private final SchemaMappingQuery query;
+    private final ValueMapperFactory<DocumentVisitorType> valueMapperFactory;
+
+    /**
+     * Creates a new {@link SchemaMapper} for a query described in queryResultTableSchema
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMapper.java:23]:
```suggestion
     * Creates a new {@link SchemaMapper} for a query described in {@code queryResultTableSchema}.
```

Comment (morazow) [reply src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMapper.java:23]: But I could not find `queryResultTableSchema`, did you mean `query` here?

File: src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathIteratorFactory.java (hunk 1 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathIteratorFactory.java:44]: The default onw is:
````
return Spliterators.spliteratorUnknownSize(this.iterator(), 0);
```
That fits here

File: src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMapper.java (hunk 2 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMapper.java]: yes. I fixed the comment.

**Pull request merged**